nerfstudio/fields/nerfacto_field.py

```diff
@@ -182,6 +190,21 @@ def __init__(
                 in_dim=self.mlp_semantics.n_output_dims, num_classes=num_semantic_classes
             )
 
+        # predicted normals
+        if self.use_pred_normals:
+            self.mlp_pred_normals = tcnn.Network(
+                n_input_dims=self.geo_feat_dim + (3 * 2 * 2),
```

Comment:
`(3 * 2 * 2)` looks hardcoded? Can you use `self.position_encoding.n_output_dims` instead? (assuming that's correct)